### PR TITLE
Fix pattern issues

### PR DESCRIPTION
**Pull request by stevenwjy**
It turns out that the validator for `openapi-backend` couldn't understand `\p{L}` and `\p{Nd}`; hence, I changed them to `\w` in the patterns for `FirstName`, `LastName`, `MiddleName`.

### CODE REVIEW

**Comment by kleyow** [comment]
I'd like to add that `openapi-backend` might be really strict with it's OpenApi spec. I found that if an object has a `$ref` and has siblings it will throw off the validator. (Which is my fault the OA3 spec mandates that `$ref` can't have siblings).

re: regex issues. That's very strange...Were you able to find a related github issue?

**Comment by stevenwjy** [comment]
> I'd like to add that `openapi-backend` might be really strict with it's OpenApi spec. I found that if an object has a `$ref` and has siblings it will throw off the validator. (Which is my fault the OA3 spec mandates that `$ref` can't have siblings). Though not sure if this discovery applies but just throwing it out there.

Yeah, I also find it a little bit too strict, but I guess its because `openapi-backend` relies on `openapi-schema-validation`, which internally only allow each object to be either a reference or an object by itself, but not both.

> re: regex issues. That's very strange...Were you able to find a related Github issue?

Not sure about this... I haven't found one, it just appeared in the testing process. I think it could be easily reproduced once #8 gets merged.

File: v1.0/openapi3/schemas/AuthenticationType.yaml [hunk 1]
```diff
@@ -3,4 +3,6 @@ type: string
 enum:
   - OTP
   - QRCODE
+  - U2F
```
Comment (kleyow) [inline v1.0/openapi3/schemas/AuthenticationType.yaml:6]: Woah now. Mojaloop API v1.0 is set in stone. We can't add a new enumerator like that.
API v1.1 is supposed to have the U2F change (I believe).
We, as the pisp stream will have to define out own `AuthenticationType` for now in pisp api specifications.

I've been thinking of a pisp-flavoured folder in the interim but please undo any major changes to in this PR.

Comment (elnyry-sam-k) [reply v1.0/openapi3/schemas/AuthenticationType.yaml:6]: Thanks Kevin.

Yes, it will happen in v2.0.. (v1.1 was published in May 2020)

Comment (kleyow) [reply v1.0/openapi3/schemas/AuthenticationType.yaml:6]: Ah awesome. I should start adding v1.1 then.

Comment (stevenwjy) [reply v1.0/openapi3/schemas/AuthenticationType.yaml:6]: Ohh okay, sorry didn't intend to include this change in the PR and just realized that it is there...  Will update this PR soon

**Comment by stevenwjy** [comment]
Just tried to reproduce the error that I encountered with the original pattern.

The validation in `openapi-backend` resulted in the following error:
```
[
  {
    keyword: 'pattern',
    dataPath: '.requestBody.party.personalInfo.complexName.firstName',
    schemaPath: '#/properties/requestBody/properties/party/properties/personalInfo/properties/complexName/properties/firstName/pattern',
    params: { pattern: "^(?!\\s*$)[\\p{L}\\p{Nd} .,'-]{1,128}$" },
    message: `should match pattern "^(?!\\s*$)[\\p{L}\\p{Nd} .,'-]{1,128}$"`
  },
  [length]: 1
]
```

The value for the `firstName` field that failed the validation was "Reinhold", which is supposed to be a valid party name.

File: v1.0/openapi3/schemas/AuthenticationType.yaml (hunk 1 above)
Comment (stevenwjy) [reply v1.0/openapi3/schemas/AuthenticationType.yaml]: Reverted this update

**Comment by kleyow** [comment]
https://github.com/mojaloop/api-snippets/pull/1#discussion_r443460855
@stevenwjy @elnyry-sam-k @rmothilal Hey Rajiv. I can't remember the rationale behind the regex change back in #1 (I don't think I questioned it at the time)

Steven is encountering problems with the change. Can you refresh my memory as to why the regex deviated from v1.0

It's also deviated from v1.1 https://github.com/mojaloop/mojaloop-specification/blob/master/documents/v1.1-document-set/fspiop-v1.1-openapi3.yaml#L2288

**Comment by kleyow** [comment]
Related  @rmothilal @stevenwjy 
https://github.com/mojaloop/mojaloop-specification/issues/56
https://github.com/mojaloop/project/issues/1261
https://github.com/mojaloop/project/issues/1457

Seems Rajiv has an idea of the problem you are facing https://gitter.im/ajv-validator/ajv?at=5e85ee5a9d51802235e37a8e

I'm in favor of reverting it for now. Not many people are using `api-snippets` and I don't want to block your work.

Please run `npm run buildv1.0` to update the doc.

**Comment by stevenwjy** [comment]
> Seems Rajiv has an idea of the problem you are facing https://gitter.im/ajv-validator/ajv?at=5e85ee5a9d51802235e37a8e

I see, yeah, I think the issue hasn't been resolved so far in `ajv`, perhaps do you want to create a tracking issue for this?
 
> I'm in favor of reverting it for now. Not many people are using api-snippets and I don't want to block your work.

Okay, got it.

> Please run npm run buildv1.0 to update the doc.

Shall I run npm run buildv1.1 as well?

**Comment by kleyow** [comment]
> I see, yeah, I think the issue hasn't been resolved so far in ajv, perhaps do you want to create a tracking issue for this?

I will.

> Shall I run npm run buildv1.1 as well?

Sure but don't think 1.1 has been updated.

**Comment by lewisdaly** [comment]
Hmm we need to be careful here - @rmothilal has been doing a lot of work on these regex expressions.

I'm happy to merge this to not block you for now @stevenwjy (since we aren't using the API Snippets much elsewhere) but let's be careful here

**Comment by kleyow** [comment]
Yea Lewis. Seem like it's just the problem with `ajv`. Rajiv seems to be tracking the issue. 

Seems like we need to maybe fix the issue in `ajv` and then openapi-backend. (By we I mean maybe me :))

**Comment by stevenwjy** [comment]
Okay, updated. I think the issues caused by the changes in this PR only occur when we have to deal with unicode characters. Aside from that, I don't think there should be any significant impact.

**Pull request merged**